contracts/core/Registry.sol

```diff
@@ -106,125 +127,176 @@ contract Registry is IRegistry, Native, AccessControl, Transfer {
         profilesById[profileId] = profile;
         anchorToProfileId[profile.anchor] = profileId;
 
-        // assign roles
+        // Assign roles for the profile members
         uint256 memberLength = _members.length;
         for (uint256 i = 0; i < memberLength;) {
             address member = _members[i];
+
+            // Will revert if any of the addresses are a zero address
             if (member == address(0)) {
                 revert ZERO_ADDRESS();
             }
+
+            // Grant the role to the member and emit the event for each member
             _grantRole(profileId, member);
             unchecked {
                 i++;
             }
         }
 
+        // Emit the event that the profile was created
         emit ProfileCreated(profileId, profile.nonce, profile.name, profile.metadata, profile.owner, profile.anchor);
 
+        // Return the profileId
         return profileId;
     }
 
     /// @notice Updates the name of the profile and generates new anchor
+    ///
+    /// Requirements: Must be the owner of the profile
+    ///
+    /// Note: Use caution when updating your profile name as it will generate a new anchor address
+    /// Note: You can always update the name back to the original name to get the original anchor address
+    ///
     /// @param _profileId The profileId of the profile
     /// @param _name The new name of the profile
-    /// @dev Only owner can update the name.
+    /// @dev Only owner can update the name
     function updateProfileName(bytes32 _profileId, string memory _name)
         external
         onlyProfileOwner(_profileId)
         returns (address)
     {
+        // TODO: Do we want to check if the string is empty? @thelostone-mc @kurtmerbeth
+        // Generate a new anchor address
         address anchor = _generateAnchor(_profileId, _name);
 
+        // Get the profile using the profileId from the mapping
         Profile storage profile = profilesById[_profileId];
+
+        // Set the new name
         profile.name = _name;
 
-        // remove old anchor
+        // Remove old anchor
+        // TODO: Why do we set to address zero to only update again with new anchor? @thelostone-mc @kurtmerbeth
```

Comment:
we do not update _with_ the anchor again

Its a anchor => profile mappping

when creating a new anchor, the `old anchor` loses the link to the profile and the the link is set to the `new anchor`